src/braket/experimental/autoqasm/program/program.py

```diff
@@ -43,6 +43,24 @@ def _get_local() -> threading.local:
     return _local
 
 
+@dataclass
+class SerializationConfig:
+    simplify_constants: Optional[bool] = False
+    """Whether to simplify constants, such as pi, from numerics to symbols. Defaults to False."""
+
+    auto_defcalgrammar: Optional[bool] = False
+    """Whether to automatically include defcalgrammar when pulses are used. Default to False."""
+
+    include_externs: Optional[bool] = False
+    """Whether to include externs. Default to False."""
+
+    return_capture_to_bitvar: Optional[bool] = True
+    """Whether to assign the return of capture_v0 to a bit variable. Default to True. """
+
+    _encal_declarations: Optional[bool] = field(init=False)
+    """Oqpy's encal_declarations option. AutoQASM sets it to True if a program includes pulse."""
```

Comment:
Thought it would be neat to integrate that in the serialization config, but yes it's indeed not a user config. I can flip it back to `has_pulse_control`